rocAL/rocAL_pybind/example/new_api/rocAL_api_coco_pipeline_test.py

```diff
@@ -0,0 +1,281 @@
+        self.p = (1 if color_format is types.GRAY else 3)
+        self.out = np.empty(
+                (self.bs*self.n,int(self.h/self.bs), self.w,self.p), dtype="ubyte")
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        print("In the next routine of COCO Iterator")
+        if(self.loader.isEmpty()):
+            timing_info = self.loader.Timing_Info()
+            print("Load     time ::", timing_info.load_time)
+            print("Decode   time ::", timing_info.decode_time)
+            print("Process  time ::", timing_info.process_time)
+            print("Transfer time ::", timing_info.transfer_time)
+            raise StopIteration
+
+        if self.loader.run() != 0:
+            raise StopIteration
+        self.lis = []  # Empty list for bboxes
+        self.lis_lab = []  # Empty list of labels
+
+         #Copy output from buffer to numpy array
+        self.loader.copyImage(self.out)
+#Image id of a batch of images
+        self.image_id = np.zeros(self.bs, dtype="int32")
+        self.loader.GetImageId(self.image_id)
+# Count of labels/ bboxes in a batch
+        self.bboxes_label_count = np.zeros(self.bs, dtype="int32")
+        self.count_batch = self.loader.GetBoundingBoxCount(
+            self.bboxes_label_count)
+        print("Count Batch:", self.count_batch)
+# 1D labels & bboxes array
+        self.encoded_bboxes = np.zeros((self.count_batch*4), dtype="float32")
+        self.encoded_labels = np.zeros(self.count_batch, dtype="int32")
+        self.loader.copyEncodedBoxesAndLables(
+            self.encoded_bboxes, self.encoded_labels)
+# Image sizes of a batch
+        self.img_size = np.zeros((self.bs * 2), dtype="int32")
+        self.loader.GetImgSizes(self.img_size)
+        print("Image sizes:", self.img_size)
+
+        encoded_bboxes_tensor = torch.tensor(self.encoded_bboxes).view(self.bs, -1, 4).contiguous()
+        encodded_labels_tensor=  torch.tensor(self.encoded_labels).long().view(self.bs, -1)
+        image_id_tensor = torch.tensor(self.image_id)
+        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
+        for i in range(self.bs):
+            index_list =[ ]
+            actual_bboxes=[]
+            actual_labels =[]
+            for idx, x in enumerate(encodded_labels_tensor[i]):
+                if x!=0:
+                    index_list.append(idx)
+                    actual_bboxes.append(encoded_bboxes_tensor[i][idx].tolist())
+                    actual_labels.append(encodded_labels_tensor[i][idx].tolist())
+
+            if self.display!="False":
+                if self.n == 1:
+                    img = torch.from_numpy(self.out)
+                    draw_patches(img[i], self.image_id[i], actual_bboxes, self.path)
+                else:
+                    img = torch.from_numpy(self.out)
+                    for idx in range(self.n):
+                        draw_all_images(img[i],self.image_id[i],idx, self.path)
```

Comment:
The definition of function taking only 3 arguments. Has this been tested?

Reply:
There were 2 functions for the same functionality, removed one